test-tools/issuer-front-end/src/Main.tsx

```diff
@@ -91,10 +91,75 @@ async function addRevokationKey(
 }
 
 type AttributeDetails = { tag: string; type: string; value: string | undefined; required: boolean };
+
 const WRONG_ATTRIBUTES: AttributeDetails[] = [
     { tag: 'myWrongAttribute', type: 'string', value: 'myWrongValue', required: false },
 ];
 
+function renderAddPrompt(details: AttributeDetails) {
+    if (details.required) {
+        return (
+            <div>
+                {' '}
+                Add <b className="text-warning">required</b> attribute <strong> {details.tag} </strong>{' '}
+            </div>
+        );
+    }
+    return (
+        <div>
+            {' '}
+            Add <strong> {details.tag} </strong>{' '}
+        </div>
+    );
+}
+
+async function extractFromSchema(url: string): Promise<AttributeDetails[]> {
+    const response = await fetch(url);
+    if (!response.ok) {
+        throw new Error(`Unable to get schema. Response code: ${response.status}`);
+    }
+    const json = await response.json();
+    const { properties, required } = json.properties.credentialSubject.properties.attributes;
+
+    const attributeSchemaValues: AttributeDetails[] = [];
+    Object.entries(properties).forEach(([key, obj]) => {
+        attributeSchemaValues.push({
+            tag: key,
+            type: (obj as { type: string }).type,
+            value: undefined,
+            required: (required as string[]).includes(key),
+        });
+    });
+    return attributeSchemaValues;
+}
+
+function parseAttributesFromForm(
+    attributeSchema: AttributeDetails[],
+    setParsingError: (msg: string) => void
+): Attribute {
+    const attributes: Attribute = {};
+    attributeSchema.forEach((obj) => {
+        if (obj.required && obj.value === undefined) {
+            setParsingError(`Attribute ${obj.tag} needs to be set.`);
```

Comment:
I suggest not checking if all required attributes are set. That way we can test the wallet if it correctly prevents us from adding a credential if a required attribute is missing. 

```
        if (obj.value !== undefined) {
            if (obj.type === 'string') {
                attributes[obj.tag] = obj.value;
            } else if (obj.type === 'number') {
                attributes[obj.tag] = BigInt(obj.value);
            } else {
                setParsingError(
                    `Attribute ${obj.tag} has type ${obj.type}. Only the types string/number are supported.`
                );
                throw new Error(
                    `Attribute ${obj.tag} has type ${obj.type}. Only the types string/number are supported.`
                );
            }
        }```

Reply:
Good point.

Reply:
I have added this, but I also added a 
```
console.warn(`Attribute ${obj.tag} is required but has not been set.`);
```

so that we can see during testing what has happened.